src/control/tools/BaseStrokeHandler.h

```diff
@@ -12,14 +12,16 @@
 #pragma once
 
 #include "model/Point.h"
+#include "model/Snapping.h"
 #include "view/DocumentView.h"
 
 #include "InputHandler.h"
+#include "SnapToGridInputHandler.h"
 
 enum DIRSET_MODIFIERS { NONE = 0, SET = 1, SHIFT = 1 << 1, CONTROL = 1 << 2 };
 
 
-class BaseStrokeHandler: public InputHandler {
+class BaseStrokeHandler: public InputHandler, public SnapToGridInputHandler {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.